rosidl_typesupport_introspection_cpp/include/rosidl_typesupport_introspection_cpp/message_introspection.hpp

```diff
@@ -27,31 +27,68 @@
 namespace rosidl_typesupport_introspection_cpp
 {
 
+/// Structure used to describe a single field of an interface type.
 typedef struct ROSIDL_TYPESUPPORT_INTROSPECTION_CPP_PUBLIC MessageMember_s
 {
+  /// The name of the field.
   const char * name_;
+  /// The type of the field as a value of the field types enum,
+  /// e.g.rosidl_typesupport_introspection_cpp::ROS_TYPE_FLOAT
   uint8_t type_id_;
+  /// If the field is a string, the upper bound on the length of the string.
   size_t string_upper_bound_;
+  /// If the type_id_ value is rosidl_typesupport_introspection_cpp::ROS_TYPE_MESSAGE
+  /// this points to an array describing the fields of the sub-interface.
   const rosidl_message_type_support_t * members_;
+  /// True if this field is an array, false if it is a unary type. An array has the same value for
+  /// type_id_.
   bool is_array_;
+  /// If is_array_ is true, this contains the number of members in the array.
   size_t array_size_;
+  /// If is_array_ is true, this specifies if the array has a maximum size. If it is true, the
+  /// value in array_size_ is the maximum size.
   bool is_upper_bound_;
+  /// The bytes into the interface's in-memory representation that this field can be found at.
   uint32_t offset_;
+  /// If the interface has a default value, this points to it.
   const void * default_value_;
+  /// If is_array_ is true, a pointer to a function that gives the size of array members.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
   size_t (* size_function)(const void *);
+  /// If is_array_ is true, a pointer to a function that gives a const pointer to the member of the
+  /// array indicated by index.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
   const void * (*get_const_function)(const void *, size_t index);
+  /// If is_array_ is true, a pointer to a function that gives a pointer to the member of the
+  /// array indicated by index.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
   void * (*get_function)(void *, size_t index);
+  /// If is_array_ is true, a pointer to a function that resizes the array.
+  /// First argument should be a pointer to the actual memory representation of the member.
+  /// Only used for nested interface types.
   void (* resize_function)(void *, size_t size);
 } MessageMember;
 
+/// Structure used to describe all fields of a single interface type.
 typedef struct ROSIDL_TYPESUPPORT_INTROSPECTION_CPP_PUBLIC MessageMembers_s
 {
+  /// The namespace in which the interface resides, e.g. "example_messages__msg" for
```

Comment:
When `rosidl_typesupport_introspection_c` is used, it will be `example_message__msg`. When `rosidl_typesupport_introspection_cpp` is used, it will be `example_message::msg`. Maybe that difference could be standardized, but it seems to be that at the moment.

Example generated code from latest ROS 2

```c++
static const ::rosidl_typesupport_introspection_cpp::MessageMembers JointState_message_members = {
  "sensor_msgs::msg",  // message namespace
  "JointState",  // message name
  5,  // number of fields
  sizeof(sensor_msgs::msg::JointState),
  JointState_message_member_array,  // message members
  JointState_init_function,  // function to initialize message memory (memory has to be allocated)
  JointState_fini_function  // function to terminate message instance (will not free memory)
};
```

```C
static const rosidl_typesupport_introspection_c__MessageMembers sensor_msgs__msg__JointState__rosidl_typesupport_introspection_c__JointState_message_members = {
  "sensor_msgs__msg",  // message namespace
  "JointState",  // message name
  5,  // number of fields
  sizeof(sensor_msgs__msg__JointState),
  sensor_msgs__msg__JointState__rosidl_typesupport_introspection_c__JointState_message_member_array,  // message members
  sensor_msgs__msg__JointState__rosidl_typesupport_introspection_c__JointState_init_function,  // function to initialize message memory (memory has to be allocated)
  sensor_msgs__msg__JointState__rosidl_typesupport_introspection_c__JointState_fini_function  // function to terminate message instance (will not free memory)
};
```

Reply:
Good catch, I added both to the comment ba5dd8d493c438ad8424d966a6c199dd14169462